h/isgenq.h

```diff
@@ -41,26 +41,137 @@ typedef struct ENQToken_tag {
 
 #pragma map(isgenqGetExclusiveLockOrFail, "ENQGETXF")
 #pragma map(isgenqGetExclusiveLock, "ENQGETX")
+#pragma map(isgenqGetSharedLockOrFail, "ENQGETSF")
+#pragma map(isgenqGetSharedLock, "ENQGETS")
+#pragma map(isgenqTestExclusiveLock, "ENQTSTX")
+#pragma map(isgenqTestSharedLock, "ENQTSTS")
 #pragma map(isgenqTestLock, "ENQTEST")
 #pragma map(isgenqReleaseLock, "ENQREL")
 
+/**
+ * @brief The function acquires an exclusive lock or fails if a lock is already
+ * being held for this QNAME and RNAME combination. See more details in the
+ * ISGENQ documentation.
+ *
+ * @param qname Enqueue major name
+ * @param rname Enqueue minor name
+ * @param scope Scope of the enqueue (use one of ISGENQ_SCOPE_xxxx)
+ * @param token Token used to release the enqueue
+ * @param reasonCode Reason code from ISGENQ
+ * @return Return code from ISGENQ
+ */
 int isgenqGetExclusiveLockOrFail(const QName *qname,
                                  const RName *rname,
                                  uint8_t scope,
                                  ENQToken *token,
                                  int *reasonCode);
-
+/**
+ * @brief The function acquires an exclusive lock. If a lock is already being
+ * held for this QNAME and RNAME combination, the current task is suspended.
+ * See more details in the ISGENQ documentation.
+ *
+ * @param qname Enqueue major name
+ * @param rname Enqueue minor name
+ * @param scope Scope of the enqueue (use one of ISGENQ_SCOPE_xxxx)
+ * @param token Token used to release the enqueue
+ * @param reasonCode Reason code from ISGENQ
+ * @return Return code from ISGENQ
+ */
 int isgenqGetExclusiveLock(const QName *qname,
                            const RName *rname,
                            uint8_t scope,
                            ENQToken *token,
                            int *reasonCode);
-
+/**
+ * @brief The function acquires a shared lock or fails if a lock is already
+ * being held for this QNAME and RNAME combination. See more details in the
+ * ISGENQ documentation.
+ *
+ * @param qname Enqueue major name
+ * @param rname Enqueue minor name
+ * @param scope Scope of the enqueue (use one of ISGENQ_SCOPE_xxxx)
+ * @param token Token used to release the enqueue
+ * @param reasonCode Reason code from ISGENQ
+ * @return Return code from ISGENQ
+ */
+int isgenqGetSharedLockOrFail(const QName *qname,
```

Comment:
I understand it's verbose, although I don't find it confusing - you either obtain the lock or the request fails and you get a non-zero RC.

I like the proposed name but if I use it the new names will not be consistent with the existing "exclusive lock" functions.

Reply:
Semantics! I find it confusing because the word "fail" may have a different connotation. It could be interpreted as it will  result in an ABEND or stop processing. 

I accept your point that consistency is important.

Reply:
I know this is an annoying, pedantic nitpick but maybe you could change all the ```orFail``` function names to ```try```? :)

Reply:
I can't rename ```isgenqGetExclusiveLockOrFail```, there are users of that function. Do you want me to make an alias?

Reply:
Nope. If this is already a published API it's too late to change now.